### PR TITLE
use a matrix strategy to build with Qt 5.14

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -33,6 +33,11 @@ jobs:
 
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        # tests run on 20.04 (Qt 5.12), compile test on 20.10 (Qt 5.14)
```
Comment (nyalldawson) [inline .github/workflows/run-tests.yml:38]: I'd be interested to see if we could enable tests on 5.14 too... I think most would pass now.

Comment (lbartoletti) [reply .github/workflows/run-tests.yml:41]: You mean, Qt 5.15?

Comment (3nids) [reply .github/workflows/run-tests.yml:41]: Ubuntu 20.10 has Qt 5.14
With this new workflow, it should be much easier to test with new Qt versions.
@nyalldawson as soon as I have this merged, I'll make a new PR to see what happens with the tests. We could easily have a skip tests file depending on the Qt version later on. That should also ease the move to newer Qt/base image for the future 🤞

**Comment by 3nids** [comment]
not sure what is happening here: `undefined reference to `Qt3DExtras::Qt3DWindow::Qt3DWindow(QScreen*)'`

Currently, the build with Qt 5.14 is running well on the workflow based on Fedora 32: https://github.com/qgis/QGIS/tree/master/.ci/qt5_14

In this PR, I am trying to build on Ubuntu 20.10 (Qt 5.14.2).

I have found something related: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=960954

Anyone understands? 
ping @m-kuhn @PeterPetrik

**Comment by 3nids** [comment]
not sure what is happening here: `undefined reference to `Qt3DExtras::Qt3DWindow::Qt3DWindow(QScreen*)'`

Currently, the build with Qt 5.14 is running well on the workflow based on Fedora 32: https://github.com/qgis/QGIS/tree/master/.ci/qt5_14

In this PR, I am trying to build on Ubuntu 20.10 (Qt 5.14.2).

I have found something related: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=960954

Anyone understands? 
ping @m-kuhn @PeterPetrik

**Comment by 3nids** [comment]
not sure what is happening here: `undefined reference to `Qt3DExtras::Qt3DWindow::Qt3DWindow(QScreen*)'`

Currently, the build with Qt 5.14 is running well on the workflow based on Fedora 32: https://github.com/qgis/QGIS/tree/master/.ci/qt5_14

In this PR, I am trying to build on Ubuntu 20.10 (Qt 5.14.2).

I have found something related: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=960954

Anyone understands? 
ping @m-kuhn @PeterPetrik

**Comment by PeterPetrik** [comment]
@wonder-sk you run on Ubuntu 20.x, don't you?

**Pull request merged**